synopsys-task/src/main.ts

```diff
@@ -41,25 +43,20 @@ export async function run() {
   } catch (error: any) {
     throw error;
   } finally {
-    const isPullRequest = isPullRequestEvent();
     if (parseToBoolean(inputs.BLACKDUCK_REPORTS_SARIF_CREATE)) {
-      if (!isPullRequest) {
-        console.log("BLACKDUCK_REPORTS_SARIF_CREATE is enabled");
-        uploadSarifResultAsArtifact(
-          constants.DEFAULT_BLACKDUCK_SARIF_GENERATOR_DIRECTORY,
-          inputs.BLACKDUCK_REPORTS_SARIF_FILE_PATH
-        );
-      }
+      console.log("BLACKDUCK_REPORTS_SARIF_CREATE is enabled");
```

Comment:
I believe, here we should check if pr comment enabled flag before uploading sarif result.

Reply:
We are checking the flag with this if condition in the previous line - 
`if (parseToBoolean(inputs.BLACKDUCK_REPORTS_SARIF_CREATE))`

I have removed the `isPullRequestEvent` check because we don't need to check the event to upload the sarif file. It will only be generated for non-PR context and for PR context we are not passing the sarif flag to bridge. So even if the user passes sarif flag for PR context, here it will try to upload but the file will not be generated and that scenario is handled in this method - 
```
uploadSarifResultAsArtifact(
        constants.DEFAULT_BLACKDUCK_SARIF_GENERATOR_DIRECTORY,
        inputs.BLACKDUCK_REPORTS_SARIF_FILE_PATH
      );
```

We cannot check the pull request event here because `AzurePrResponse` from the API call is not available in the main.ts and if we want to get the `AzurePrResponse` here then there will be a lot of changes since we will then need to return the `AzurePrResponse` from tools-parameter.ts to main.ts and there are multiple ts file involved there.

Reply:
As per workflow simplification, prcomment will always be set as true and we are ignoring prcomment based on `isPullRequestEvent` check.

The `isPullRequestEvent` check was here also but to check the `isPullRequestEvent` we need to pass the `AzurePrResponse` as argument to this method and `AzurePrResponse` is not available in main.ts file. And if we want to get the `AzurePrResponse` from the tools-parameter.ts to main.ts then there will be a lot of code changes to achieve this because there are multiple ts file involved between tools-parameter.ts and main.ts file.

Therefore if we check the prcomment enabled flag here and it will always be true, so there will be scenarios when sarif report will not be uploaded though it should be uploaded.

SARIF report will only be generated for PR context, so for non-PR context when `uploadSarifResultAsArtifact()` method will be called there will be no sarif file generated and this scenarios is handled in uploadSarifResultAsArtifact() method, an info message is shown for this scenario.

Reply:
Updated code for checking PR event before uploading